cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateNativeImageTask.java

```diff
@@ -0,0 +1,131 @@
+import io.ballerina.projects.ProjectKind;
+import io.ballerina.projects.internal.model.Target;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static io.ballerina.cli.launcher.LauncherUtils.createLauncherException;
+import static io.ballerina.cli.utils.FileUtils.getFileNameWithoutExtension;
+import static io.ballerina.projects.util.ProjectConstants.BLANG_COMPILED_JAR_EXT;
+
+public class CreateNativeImageTask implements Task {
+    private final transient PrintStream out;
+    private Path output;
+    private Path currentDir;
+
+    public CreateNativeImageTask(PrintStream out, String output) {
+        this.out = out;
+        if (output != null) {
+            this.output = Paths.get(output);
+        }
+    }
+
+    @Override
+    public void execute(Project project) {
+
+        this.out.println();
+        this.out.println("Generating executable with native-image");
+
+        Path nativeDirectoryPath;
+        String nativeImageName;
+
+        Target target;
+
+        try {
+            if (project.kind().equals(ProjectKind.BUILD_PROJECT)) {
+                target = new Target(project.targetDir());
+            } else {
+                target = new Target(Files.createTempDirectory("ballerina-cache" + System.nanoTime()));
+                target.setOutputPath(getExecutablePath(project));
+            }
+        } catch (IOException e) {
+            throw createLauncherException("unable to resolve target path : " + e.getMessage());
+        } catch (ProjectException e) {
+            throw createLauncherException("unable to create executable : " + e.getMessage());
+        }
+
+        Path executablePath;
+        try {
+            executablePath = target.getExecutablePath(project.currentPackage()).toAbsolutePath().normalize();
+        } catch (IOException e) {
+            throw createLauncherException("unable to resolve executable path : " + e.getMessage());
+        }
+
+        if (project.kind().equals(ProjectKind.SINGLE_FILE_PROJECT)) {
+            nativeDirectoryPath = project.sourceRoot().toAbsolutePath().getParent().resolve("native");
+            nativeImageName = project.sourceRoot().toAbsolutePath().getParent().getFileName().toString();
+        } else {
+            nativeDirectoryPath = target.path().toAbsolutePath().resolve("native");
+            nativeImageName = project.currentPackage().packageName().toString();
+        }
+
+        try {
+            // Create native directory
+            if (!Files.exists(nativeDirectoryPath)) {
+                Files.createDirectory(nativeDirectoryPath);
+            }
+
+            String[] command = {
+                    "native-image",
+                    "-jar",
+                    executablePath.toString(),
+                    nativeDirectoryPath + File.separator + nativeImageName,
+                    "-H:MaxDuplicationFactor=4.0",
```

Comment:
We don't need to add this flag, since this will be added by ballerina runtime

Reply:
Removed max duplication factor